eelbrain/_stats/testnd.py

```
@@ -2194,7 +2206,9 @@ def add_original(self, stat_map):
         self.has_original = True
         self.dt_original = self._t0 - self._init_time
         self._original_param_map = stat_map
-        if self.samples and n_clusters:
+        if self.force_permutation and not self.samples:
+            raise ValueError("Cannot force permutations unless samples is non-zero.")
```

I would make that check in `__init__`, the earlier you can raise an error for invalid arguments the better